app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -220,5 +222,9 @@ def action_permission
         fail ::Foreman::Exception.new(N_("unknown permission for %s"), "#{params[:controller]}##{params[:action]}")
       end
     end
+
+    def remove_or_add_subscriptions_api?
```

Comment:
I see where you're going here and I like it. What about just adding a new parameter:

```
def index_response(reload_host = false)

end
```

And update `add_subscriptions`, `remove_subscriptions`, and `auto_attach` methods to call this method with `true`? Another benefit is that if anyone adds a new endpoint in the controller and they want to call `index_subscriptions` they will have to think about whether the host needs to be reloaded or not - so maybe we'll avoid this kind of bug in the future.

That way we don't need to maintain this logic.

Reply:
Looks great. Can you also add the (true) to `auto_attach` ? This method also attaches subscriptions and I suspect it has the same bug